link.go

```diff
@@ -223,6 +223,9 @@ func attachLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
 }
 
 func (l *link) addUnsettled(msg *Message) {
+	if len(msg.DeliveryTag) == 0 {
```

Comment:
FYI I got confirmation that messages coming in already settled omit the delivery tag.